debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java

```diff
@@ -126,54 +124,53 @@
                                                                  INSERT_POSTGIS_TYPES_STMT, INSERT_POSTGIS_ARRAY_TYPES_STMT));
 
     protected List<SchemaAndValueField> schemasAndValuesForNumericType() {
-        return Arrays.asList(new SchemaAndValueField("si", SchemaBuilder.OPTIONAL_INT16_SCHEMA, (short) 1),
+        final List<SchemaAndValueField> fields = new ArrayList<SchemaAndValueField>(
+                Arrays.asList(new SchemaAndValueField("si", SchemaBuilder.OPTIONAL_INT16_SCHEMA, (short) 1),
                              new SchemaAndValueField("i", SchemaBuilder.OPTIONAL_INT32_SCHEMA, 123456),
                              new SchemaAndValueField("bi", SchemaBuilder.OPTIONAL_INT64_SCHEMA, 1234567890123L),
                              new SchemaAndValueField("r", Schema.OPTIONAL_FLOAT32_SCHEMA, 3.3f),
                              new SchemaAndValueField("db", Schema.OPTIONAL_FLOAT64_SCHEMA, 4.44d),
+                             new SchemaAndValueField("r_nodec", Schema.OPTIONAL_FLOAT32_SCHEMA, 3.0f),
+                             new SchemaAndValueField("db_nodec", Schema.OPTIONAL_FLOAT64_SCHEMA, 4.0d),
                              new SchemaAndValueField("ss", Schema.INT16_SCHEMA, (short) 1),
                              new SchemaAndValueField("bs", Schema.INT64_SCHEMA, 123L),
-                             new SchemaAndValueField("b", Schema.OPTIONAL_BOOLEAN_SCHEMA, Boolean.TRUE));
-    }
-
-    protected List<SchemaAndValueField> schemasAndValuesForFpTypeWithoutDecimals() {
-        return Arrays.asList(new SchemaAndValueField("si", SchemaBuilder.OPTIONAL_INT16_SCHEMA, (short) 1),
-                             new SchemaAndValueField("i", SchemaBuilder.OPTIONAL_INT32_SCHEMA, 123456),
-                             new SchemaAndValueField("bi", SchemaBuilder.OPTIONAL_INT64_SCHEMA, 1234567890123L),
-                             new SchemaAndValueField("r", Schema.OPTIONAL_FLOAT32_SCHEMA, 3.0f),
-                             new SchemaAndValueField("db", Schema.OPTIONAL_FLOAT64_SCHEMA, 4.0d),
-                             new SchemaAndValueField("ss", Schema.INT16_SCHEMA, (short) 1),
-                             new SchemaAndValueField("bs", Schema.INT64_SCHEMA, 123L),
-                             new SchemaAndValueField("b", Schema.OPTIONAL_BOOLEAN_SCHEMA, Boolean.TRUE));
+                             new SchemaAndValueField("b", Schema.OPTIONAL_BOOLEAN_SCHEMA, Boolean.TRUE))
+                );
+        if (!DecoderDifferences.areSpecialFPValuesUnupported()) {
+            fields.addAll(Arrays.asList(
+                    new SchemaAndValueField("r_nan", Schema.OPTIONAL_FLOAT32_SCHEMA, Float.NaN),
+                    new SchemaAndValueField("db_nan", Schema.OPTIONAL_FLOAT64_SCHEMA, Double.NaN),
+                    new SchemaAndValueField("r_inf", Schema.OPTIONAL_FLOAT32_SCHEMA, Float.POSITIVE_INFINITY),
```

Comment:
Adding two fields for `NEGATIVE_INFINITY` would be the icing on the cake :)

Reply:
You are martinet :-) ok